google/auth/compute_engine/_metadata.py

```diff
@@ -102,6 +143,32 @@ def detect_gce_residency_linux():
     return content.startswith(_GOOGLE)
 
 
+def _prepare_request_for_mds(request, use_mtls=False):
+    """Prepares a request for the metadata server.
+
+    This will check if mTLS should be used and mount the mTLS adapter if needed.
+
+    Args:
+        request (google.auth.transport.Request): A callable used to make
+            HTTP requests.
+        use_mtls (bool): Whether to use mTLS for the request.
+
+    Returns:
+        google.auth.transport.Request: A request object to use.
+            If mTLS is enabled, the request will have the mTLS adapter mounted.
+            Otherwise, the original request will be returned unchanged.
+    """
+    if not use_mtls:
+        return request
+
+    adapter = _mtls.MdsMtlsAdapter()
+    if not request.session:
+        request.session = requests.Session()
+    for host in _GCE_DEFAULT_MDS_HOSTS:
+        request.session.mount(f"https://{host}/", adapter)
+    return request
```

Comment:
you are returning the same request object as the input. All changes are inside the object.

Do we really have to return?